modules/compute/src/integrationTest/java/org/apache/ignite/internal/compute/ItThinClientWorkerShutdownTest.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.compute;
+
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.compute.IgniteCompute;
+import org.apache.ignite.internal.app.IgniteImpl;
+import org.junit.jupiter.api.AfterEach;
+
+class ItThinClientWorkerShutdownTest extends ItWorkerShutdownTest {
+
+    private IgniteClient client;
+
+    @AfterEach
+    void cleanup() throws Exception {
+        client.close();
+    }
+
+    @Override
+    IgniteCompute compute(IgniteImpl entryNode) {
+        String address = "127.0.0.1:" + entryNode.clientAddress().port();
+        client = IgniteClient.builder().addresses(address).build();
```

Comment:
1. A new client is started on every call to this method
2. The clients are never closed, causing extra resource consumption and potential issues in other tests

Let's start one client before the test, save it to a field, and close after the test.

Reply:
We can have a map from address to client, and clean it up after the test. But we can't leave the clients running in the background.